### PR TITLE
Feat: Storybook: Improve component organisation - Overlays Category - Issue #66275

### PR DESCRIPTION
**Pull request by hbhalodia**
## What?

- Issue - #66275 
- Changes the componenet metadata to use the sitemap structure to ease user navigation to storybook.

## Why?
This change is part of the larger [Storybook Improvements](https://make.wordpress.org/design/2024/09/17/design-systems-storybook-improvements/) and the [shared Sitemap](https://www.figma.com/board/0ZfHa37xxWoMqy7McezUNX/WordPress.org-Storybook-Sitemap?node-id=126-5118&t=ruCtnVTDVjHTfksc-1). 

## How?

Adding Categories that helps organize and group components into sections based on their role and function within the library makes our library easier to navigate and understand. 

## Testing Instructions

1. Checkout to the PR.
2. Run `npm run storybook:dev`.
3. Check for the updated components with category that are changed in PR.

### Testing Instructions for Keyboard

- None

## Screenshots or screencast 
Screenshot 2024-11-01 at 9 17 16 AM
Screenshot 2024-11-01 at 9 17 23 AM

### CODE REVIEW

File: packages/components/src/confirm-dialog/stories/index.story.tsx
Comment (tyxla) [inline packages/components/src/confirm-dialog/stories/index.story.tsx]: As this component is still formally experimental, if we change its ID, it will cause the previous URL not to work:

http://localhost:50240/?path=/docs/components-experimental-confirmdialog--docs

To fix that, we have 2 options:
- Use `id: 'components-experimental-confirmdialog'`
- Add the component to `PREVIOUSLY_EXPERIMENTAL_COMPONENTS`. 

Since the component is still experimental, I'd prefer the former option. Once we stabilize the component, we can always add it to `PREVIOUSLY_EXPERIMENTAL_COMPONENTS`

File: packages/components/src/confirm-dialog/stories/index.story.tsx
Comment (tyxla) [inline packages/components/src/confirm-dialog/stories/index.story.tsx:19]: Another thing (cc @ciampo @mirka): experimental components will fall in their own section down below, meaning they will have their own Overlays directory. Is that desired? Should we consider marking the experimental components just with badges, like `CustomSelectControlV2` is marked WIP?

**Review by tyxla — COMMENTED** [review]
Thanks for working on this one @hbhalodia! 🙌 

It mostly looks good, but there are a few considerations for experimental components we might need to consider before shipping.

File: packages/components/src/confirm-dialog/stories/index.story.tsx
Comment (hbhalodia) [reply packages/components/src/confirm-dialog/stories/index.story.tsx:19]: > experimental components will fall in their own section down below, meaning they will have their own Overlays directory

Waiting on this for confirmation.

Comment (tyxla) [reply packages/components/src/confirm-dialog/stories/index.story.tsx:19]: I've been discussing this with @ciampo, and we concluded that it's a problem that can be solved separately. 

In the future we may remove the "Experimental" distinction from the Storybook component list, and those categories that we're introducing will be on the top level. That being said, we could leave things as they are and ship this PR, and I've opened #66805 to discuss this separately.

File: packages/block-library/src/image/block.json
Comment (tyxla) [inline packages/block-library/src/image/block.json]: This change looks unintentional and unrelated to the PR - could you revert it before we ship the PR?

File: packages/block-library/src/post-content/block.json
Comment (tyxla) [inline packages/block-library/src/post-content/block.json]: This change looks unintentional and unrelated to the PR - could you revert it before we ship the PR?

File: packages/block-library/src/image/block.json
Comment (hbhalodia) [reply packages/block-library/src/image/block.json]: Hi @tyxla, I would unable to do so, because when I sync the current branch with trunk and rebase it, and did commit, it was done automatically when run linting using precommit hooks.

File: packages/block-library/src/post-content/block.json
Comment (hbhalodia) [reply packages/block-library/src/post-content/block.json]: Hi @tyxla, I would unable to do so, because when I sync the current branch with trunk and rebase it, and did commit, it was done automatically when run linting using precommit hooks.

Comment (tyxla) [reply packages/block-library/src/post-content/block.json]: This is odd! It looks completely unrelated.

I'd be happy to help with it, but I'll need permissions. Do you see this in the sidebar of this PR? You might need to tick the checkbox so we have permissions to push to your branch:

Screenshot 2024-11-08 at 1 05 54

Comment (hbhalodia) [reply packages/block-library/src/post-content/block.json]: Hi @tyxla, thanks for the help, I see this box is already checked.

Comment (hbhalodia) [reply packages/block-library/src/post-content/block.json]: Hi @tyxla, I had fix this, by adding `--no-verify` and committed the changes.

Thank You,

**Review by mirka — APPROVED** [review]
🚀 

By the way @hbhalodia, now that you are part of the Gutenberg team, you should be able to push branches directly to the repo, and you'll be able to merge PRs yourself once they are approved 🙂

**Pull request merged**
